okhttp/src/main/kotlin/okhttp3/internal/http2/Http2Stream.kt

```diff
@@ -283,7 +285,10 @@ class Http2Stream internal constructor(
     val open: Boolean
     synchronized(this) {
       if (!hasResponseHeaders || !inFinished) {
-        hasResponseHeaders = true
+        val status = headers[Header.RESPONSE_STATUS_UTF8]?.toIntOrNull()
```

Comment:
Do you mind retargeting `master`, then cherry-picking this back to okhttp_4x ?

I’m a little surprised to see status codes handled at this layer

Reply:
I think it went here so I could make the minimal change.

I can try again to apply some of the changes directly, it's been a while, so it's a bit vague why it broken down.

Reply:
I’m still not a fan of handling status codes at the protocol layer. Revert this part and keep the fix for the infinite wait?

Reply:
This feels unfortunate.  I really would like to handle this properly in 4.12.